src/init.cpp

```diff
@@ -1233,6 +1233,94 @@ bool CheckHostPortOptions(const ArgsManager& args) {
     return true;
 }
 
+/**
+ * @brief Deduplicates binding vectors internally and checks for cross-vector conflicts.
+ *
+ * For vWhiteBinds: Removes duplicates with identical service AND permissions, but reports
+ * conflicts when same service has different permissions.
+ * For other vectors: Simply removes duplicates.
+ * Cross-vector duplicates are always reported as conflicts.
+ *
+ * @param vWhiteBinds Vector of whitelisted bind permissions (will be deduplicated)
+ * @param vBinds Vector of regular bind services (will be deduplicated)
+ * @param onion_binds Vector of onion-specific bind services (will be deduplicated)
+ * @return std::optional<CService> containing the first conflict found, or std::nullopt if no conflicts
+ */
+std::optional<CService> DeduplicateAndCheckBindingConflicts(
+    std::vector<NetWhitebindPermissions>& vWhiteBinds,
+    std::vector<CService>& vBinds,
+    std::vector<CService>& onion_binds)
+{
+    // Step 1: Handle vWhiteBinds deduplication with special logic for permissions
+    {
+        std::map<CService, NetPermissionFlags> seen_whitebinds;
+        std::vector<NetWhitebindPermissions> deduplicated;
+
+        for (const auto& wb : vWhiteBinds) {
+            auto it = seen_whitebinds.find(wb.m_service);
+            if (it == seen_whitebinds.end()) {
+                // First occurrence of this service
+                seen_whitebinds[wb.m_service] = wb.m_flags;
+                deduplicated.push_back(wb);
+            } else if (it->second == wb.m_flags) {
+                // Same service, same permissions - harmless duplicate, skip it
+                continue;
+            } else {
+                // Same service, different permissions - conflict!
+                return wb.m_service;
+            }
+        }
+        vWhiteBinds = std::move(deduplicated);
+    }
+
+    // Step 2: Simple deduplication for vBinds
+    {
+        std::set<CService> seen;
+        auto it = std::remove_if(vBinds.begin(), vBinds.end(),
+            [&seen](const CService& service) {
+                return !seen.insert(service).second;  // Remove if already seen
+            });
+        vBinds.erase(it, vBinds.end());
+    }
+
+    // Step 3: Simple deduplication for onion_binds
+    {
+        std::set<CService> seen;
+        auto it = std::remove_if(onion_binds.begin(), onion_binds.end(),
+            [&seen](const CService& service) {
+                return !seen.insert(service).second;  // Remove if already seen
+            });
+        onion_binds.erase(it, onion_binds.end());
```

Comment:
I find it a bit inconsistent that we're not erring here - we don't know why they provided the same thing twice, seems like a mistake, not sure we should attempt to correct it silently. Would also simplify the code if we unified the way we handle these